src/Channel.cpp

```diff
@@ -90,6 +90,47 @@ Channel::ptr_t Channel::CreateFromUri(const std::string &uri, int frame_max)
                   frame_max);
 }
 
+Channel::ptr_t Channel::CreateFromUri(const std::string &uri,
+                                      const std::string &path_to_ca_cert,
+                                      const std::string &path_to_client_key,
+                                      const std::string &path_to_client_cert,
+                                      bool verify_hostname,
+                                      int frame_max)
+{
+    amqp_connection_info info;
+    amqp_default_connection_info(&info);
+
+    boost::shared_ptr<char> uri_dup = boost::shared_ptr<char>(strdup(uri.c_str()), free);
+
+    if (0 != amqp_parse_url(uri_dup.get(), &info))
+    {
+        throw BadUriException();
+    }
+
+    if ( info.ssl )
```

Comment:
This function should only accept amqps:// URIs, it should throw something if a non-secure URI is detected.